pkg/eth/retriever_test.go

```diff
@@ -0,0 +1,344 @@
+// VulcanizeDB
+// Copyright © 2019 Vulcanize
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package eth_test
+
+import (
+	"github.com/ethereum/go-ethereum/core/types"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/vulcanize/ipfs-blockchain-watcher/pkg/eth"
+	eth2 "github.com/vulcanize/ipfs-blockchain-watcher/pkg/eth"
+	"github.com/vulcanize/ipfs-blockchain-watcher/pkg/eth/mocks"
+	"github.com/vulcanize/ipfs-blockchain-watcher/pkg/postgres"
+	"github.com/vulcanize/ipfs-blockchain-watcher/pkg/shared"
+)
+
+var _ = Describe("Retriever", func() {
+	var (
+		db        *postgres.DB
+		repo      *eth2.IPLDPublisher
+		retriever *eth2.GapRetriever
+	)
+	BeforeEach(func() {
+		var err error
+		db, err = shared.SetupDB()
```

Comment:
does this test require a real connection to PostgreSQL or we use mocks for DB as well?

Reply:
Yeah this uses a real connection to a test PostgreSQL database (vulcanize_testing)